main.go

```diff
@@ -17,35 +19,46 @@ var (
 	// less confusion.
 	storageType = getopt("STORAGE_ADAPTER", "memory")
 	numLines, _ = strconv.Atoi(getopt("NUMBER_OF_LINES", "1000"))
+	nsqHost     = getopt("DEIS_NSQD_SERVICE_HOST", "")
+	nsqPort     = getopt("DEIS_NSQD_SERVICE_PORT_TRANSPORT", "4150")
+	nsqURL      = fmt.Sprintf("%s:%s", nsqHost, nsqPort)
 )
 
 func main() {
-	syslogishServer, err := syslogish.NewServer(storageType, numLines)
+	storageAdapter, err := storage.NewAdapter(storageType, numLines)
 	if err != nil {
-		log.Fatal("Error creating syslogish server", err)
+		log.Fatal("Error creating storage adapter:", err)
 	}
-	weblogServer, err := weblog.NewServer(syslogishServer)
+	logger, err := logs.NewLogger(storageAdapter)
+	if err != nil {
+		log.Fatal("Error creating logger", err)
+	}
+
+	weblogServer, err := weblog.NewServer(logger)
 	if err != nil {
 		log.Fatal("Error creating weblog server", err)
 	}
 
-	syslogishServer.Listen()
 	weblogServer.Listen()
-
 	log.Println("deis-logger running")
 
-	// No cleanup is needed upon termination.  The signal to reopen log files (after hypothetical
-	// logroation, for instance), if applicable, is the only signal we'll care about.  Our main loop
-	// will just wait for that signal.
-	reopen := make(chan os.Signal, 1)
-	signal.Notify(reopen, syscall.SIGUSR1)
-
-	for {
-		<-reopen
-		if err := syslogishServer.ReopenLogs(); err != nil {
-			log.Fatal("Error reopening logs", err)
+	consumer.Register("logs", "consume", 30, func(message *consumer.Message) {
+		err := logger.WriteLog(message.Body)
+		if err != nil {
+			log.Printf("Unable to store message:%v\n%s", err, string(message.Body))
```

Comment:
thats not how this works. The only messages sent to nsq are those pertaining to apps deployed by deis and controller logs that also pertain to apps. We will never get component log messages in logger.